tests/testthat/test-svm-serde.R

```diff
@@ -0,0 +1,20 @@
+test_that("SVM regressor can be serialized and unserialized correctly", {
+  model <- cuml_svm(formula = mpg ~ ., data = mtcars, kernel = "rbf")
+  model_state <- cuml_serialize(model)
+
+  expected_preds <- predict(model, mtcars)
+  actual_preds <- callr::r(
+    function(model_state) {
+      library(cuml)
+
+      model <- cuml_unserialize(model_state)
+      stopifnot("cuml_svr" %in% class(model))
+      stopifnot(identical(model$mode, "regression"))
+
+      predict(model, mtcars)
+    },
+    args = list(model_state = model_state)
+  )
+
+  expect_equal(expected_preds, actual_preds)
```

Comment:
Running tests, I get this failure: 
```r
── Failure (test-svm-serde.R:19:3): SVM regressor can be serialized and unserialized correctly ──
`expected_preds` not equal to `actual_preds`.
Component “.pred”: Mean relative difference: 1
```

Inspecting the actual objects in a REPL:
```r
>   expect_equal(expected_preds, actual_preds)
Error: `expected_preds` not equal to `actual_preds`.
Component ".pred": Mean relative difference: 1
> expected_preds
# A tibble: 32 × 1
   .pred
   <dbl>
 1  20.9
 2  20.9
 3  20.5
 4  20.5
 5  18.8
 6  18.5
 7  18.5
 8  20.5
 9  20.5
10  19.1
# … with 22 more rows
> actual_preds
# A tibble: 32 × 1
   .pred
   <dbl>
 1     0
 2     0
 3     0
 4     0
 5     0
 6     0
 7     0
 8     0
 9     0
10     0
# … with 22 more rows
```

Reply:
Hmm interesting... I don't really have a reliable way to repro this, because everything is passing on my side (whether from within RSstudio IDE or otherwise... but maybe I will repro this if I repeat this test a large number of iterations (?) )

```
ttaching package: ‘rlang’

The following object is masked from ‘package:magrittr’:

    set_names

The following objects are masked from ‘package:testthat’:

    is_false, is_null, is_true

ℹ Testing cuml
✓ |  OK F W S | Context
✓ |   4       | k-means [2.1 s]                                                                      
✓ |   1       | svm-serde [4.2 s]                                                                    

══ Results ══════════════════════════════════════════════════════════════════════════════════════════
Duration: 6.3 s

[ FAIL 0 | WARN 0 | SKIP 0 | PASS 5 ]
```

Anyhow, I'll look into it and try to find out what's happening.

Reply:
When we did a screen share I noticed we were running different versions of R. Maybe try on R 4.1.1?

Reply:
I'm running R 4.1.0 at the moment for working with `{cuml}`. The older version of R you saw when we did the screen share was from inside some Conda env I believe.

Reply:
@t-kalinowski  Actually could you by any chance confirm `cuml::has_libcuml()` outputs `TRUE` for the current installation of `{cuml}` you have? If it outputs `FALSE`, then the current installation of `{cuml}` is still not linked with a valid copy of the RAPIDS cuML C++ library, and some dummy values such as all zeroes will be returned just to make CRAN happy.

Reply:
You are right! 
```
> cuml::has_libcuml()
[1] FALSE
Warning message:
In fun(libname, pkgname) : 
        The current installation of {cuml} will not function as expected
        because it was not linked with a valid version of the RAPIDS cuML shared
        library.

        To fix this issue, please follow https://rapids.ai/start.html#get-rapids
        to install the RAPIDS cuML shared library from Conda and ensure the
        'CUDA_PATH' env variable is set to a valid RAPIDS conda env directory
        (e.g., '/home/user/anaconda3/envs/rapids-21.06' or similar) during the
        installation of {cuml} or alternatively, follow
        https://github.com/yitao-li/cuml-installation-notes#build-from-source-without-conda-and-without-multi-gpu-support
        or
        https://github.com/yitao-li/cuml-installation-notes#build-from-source-without-conda-and-with-multi-gpu-support
        or similar to build and install RAPIDS cuML library from source, and
        then re-install {cuml}.

```
Currently rebuilding in the correct env. 
Maybe that warning should be emitted in .onLoad() :thinking: ?

Reply:
@t-kalinowski  Actually I thought even if `has_libcuml()` returns `FALSE`, then `expected_preds` and `actual_preds` would both contain all-zeroes and the test would still pass.

Now I'm wondering whether if you launch the test from a Conda env, `callr()` would launch a subprocess outside the Conda env and subsequently use another version of `{cuml}` that is in the host env (??). I think that is the only way for `expected_preds` to be sensible values but `actual_preds` to be all-zeroes.

Reply:
Same test failure with cuml env properly setup:
![image](https://user-images.githubusercontent.com/8462255/134077124-4146754d-bd0e-4322-b8b6-d45224dacb76.png)

loading the package with `library(cuml)` shows the warning, while `devtools::load_all()` shows no warning. 
```
> devtools::load_all(".")
ℹ Loading cuml

Attaching package: 'magrittr'

The following objects are masked from 'package:testthat':

    equals, is_less_than, not


Attaching package: 'rlang'

The following object is masked from 'package:magrittr':

    set_names

The following objects are masked from 'package:testthat':

    is_false, is_null, is_true

> cuml::has_libcuml()
[1] TRUE
```